packages/swarm-components/src/Radio.jsx

```diff
@@ -1,27 +1,61 @@
 // @flow
 import * as React from 'react';
-import Icon from './Icon';
 
 export type Props = {
-  checked: boolean,
+  /**
+   * Whether the radio button is checked.
+   */
+  checked?: boolean,
+  /**
+   * Whether the input should be interactive.
+   */
   disabled?: boolean,
+  /**
+   * A unique identifier for the input.
+   */
   id?: string,
+  /**
+   * Used to associate a group of radio buttons.
+   * Only one radio button in a group can be selected.
+   */
+  name: string,
+  /**
+   * Value of the input.
+   */
   value: string,
+  /**
+   * String label for the input. Will override any children values.
+   */
   label?: string,
-  name: string,
+  /**
+   * Use children as an alternative to the `label` prop for more complex input labels.
+   * The `label` prop will override children.
+   */
+  children: React.ReactElement<*>,
 };
 
 // Can not inline css vars as color to Icon. Gray 5 icon fill on disbaled
 const Radio = (props: Props): React.ReactElement<'label'> => {
-  const { checked, label, id = 'a', disabled, value, children, ...rest } = props;
+  const {
+    checked,
+    label,
+    id,
+    disabled,
+    value,
+    children,
+    name,
+    ...other
+  } = props;
+
   return (
-    <label data-swarm-radio={disabled ? 'disabled' : 'default'} for={id} disabled={disabled} {...rest} >
+    <label data-swarm-radio={disabled ? 'disabled' : 'default'} htmlFor={id || name} disabled={disabled}>
```

Comment:
We still have `id || name` in the `htmlFor` prop.

Reply:
In fact we don't need `htmlFor` altogether as we wrap `<input>` inside `<label>` which takes care of associating one with another.

Reply:
It looks like having these `htmlFor` attributes actually breaks checking individual radio buttons in original examples.

So, I vote for getting rid of them ;)